src/parser/datetime.rs

```diff
@@ -191,6 +379,62 @@ pub(crate) fn build_parsed_datetime(
     Ok(pdt)
 }
 
+/// Takes a 'date timezone' 'date time timezone' string and splits
+/// it into 'date {time}' and 'timezone' components
+pub(crate) fn split_timestamp_string(value: &str) -> (&str, &str) {
+    // First we need to see if the string contains " +" or " -" because timestamps can come in a format
+    // YYYY-MM-DD {+|m}<tz> (where the timezone string can have colons)
+    let mut cut = value.find(" +").unwrap_or(0);
+
+    if cut == 0 {
+        cut = value.find(" -").unwrap_or(0);
+    }
+
+    if cut > 0 {
+        let (first, second) = value.split_at(cut);
+        return (first.trim(), second.trim());
+    }
```

Comment:
this could be simpler and safer if you lean into options:

```
let cut = value.find(" +").unwrap_or_else(|| value.find(" -");
if let Some(cut) = cut {
    ..
    return cut;
}
```

I'm not actually sure what the behavior should be if the cut is at the zeroeth index, though, I would expect that to fail in interesting ways with both implementations.